compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/FirAccessorOverrideStatusTransformer.kt

```diff
@@ -119,4 +122,18 @@ internal class FirAccessorOverrideStatusTransformer(session: FirSession) :
     if (!needsOverride) return status
     return status.copy(isOverride = true)
   }
+
+  private data class Accessor(
+    val name: Name,
+    val isFunction: Boolean,
+    val returnType: ConeKotlinType
+  )
+
+  @OptIn(SymbolInternals::class)
+  private val FirCallableSymbol<*>.accessor: Accessor
+    get() = Accessor(
```

Comment:
Can we avoid the symbol internals call? Should be able to check if the symbol is an `FirNamedFunctionSymbol` or something similar

Reply:
Let me try

Reply:
You're right! Done here https://github.com/ZacSweers/metro/pull/1040/commits/98da5924e71bf404dc238d8d453953930387560d